### PR TITLE
[core] Add cpp test for array.h

### PR DESCRIPTION
**Pull request by rynewang**
The test is all compile time and static_assert.

### CODE REVIEW

File: src/ray/util/tests/array_test.cc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+              "String view array should not have unique elements");
+
+static constexpr bool index_of_int = IndexOf(std::array<int, 3>{1, 2, 3}, 2) == 1;
+static_assert(index_of_int, "Index of int should be 1");
+
+// can't test the index of not found because it's a compile error
+
+// Trivial gtest
+TEST(ArrayTest, ItRuns) {
+  // Nothing to really run because we are testing compile time behavior.
```
Comment (dentiny) [inline src/ray/util/tests/array_test.cc:49]: You don't need the dummy test case? Just a main function is enough

Comment (rynewang) [reply src/ray/util/tests/array_test.cc:49]: idk, can you have a gtest binary without test cases?

Comment (dentiny) [reply src/ray/util/tests/array_test.cc:49]: affirmatively, yes

Comment (dentiny) [reply src/ray/util/tests/array_test.cc:49]: gtest is nothing more than a registry of test classes

Comment (rynewang) [reply src/ray/util/tests/array_test.cc]: done

**Review by dentiny — APPROVED** [review]

**Pull request merged**